packages/components/src/FilterBar/context/reducer/setupFiltersState.ts

```diff
@@ -1,21 +1,45 @@
 import { Filters } from "../../types"
 import { FilterBarState, FiltersValues } from "../types"
+import { getFilterUsableState } from "../utils/getFilterUsableState"
+import { transformToFiltersState } from "../utils/transformToFiltersState"
 
 export const setupFiltersState = <ValuesMap extends FiltersValues>(
   filters: Filters<ValuesMap>,
   values: Partial<ValuesMap>
-): FilterBarState<ValuesMap> =>
-  filters.reduce<FilterBarState<ValuesMap>>(
-    (state, { Component: _, ...filter }) => {
-      state.filters[filter.id] = {
-        ...filter,
+): FilterBarState<ValuesMap> => {
+  const state = filters.reduce<FilterBarState<ValuesMap>>(
+    (baseState, { Component: _, ...filter }) => {
+      baseState.filters[filter.id] = {
         isOpen: false,
+        isUsable: true,
+        ...filter,
       }
 
       if (!filter.isRemovable || values[filter.id] !== undefined)
-        state.activeFilterIds.add(filter.id)
+        baseState.activeFilterIds.add(filter.id)
 
-      return state
+      return baseState
     },
-    { filters: {}, activeFilterIds: new Set() } as FilterBarState<ValuesMap>
+    {
+      filters: {},
+      activeFilterIds: new Set(),
+      values: {},
+    } as FilterBarState<ValuesMap>
   )
+
+  const filtersState = transformToFiltersState(state, values)
```

Comment:
How come this is a two step process? Why not extend the filter states in the first reducer?

Reply:
Terrible isn't it?

Since the consumer should be able to make a conditional out of the current state and any of its attributes, I need to combine state and values (which may be changed from outside, so not the one from state) together to then pass into `isUsableWhen` to get an accurate calculation.

Using the result from that, I then have to update the state to actually reflect what it should be (eg. deactivate and clear the value for a filter because it's actually now no longer usable).

That being said, I moved things around so many times that if duplicating everything into a single state makes more sense I could look into it. I'll just need to filter out some attributes to the user (eg. I can't have them calling `isUsableWhen` from `state` in their `isUsableWhen` cos I'll end up in a loop).

To briefly explain why some things have been split out:
- `activeFilterIds` is required to keep the sorting order for rendering. I had opted to calculate the value instead of duplicating the attribute into the state previously to prevent mismatching values.
- `state.values` has been introduced as `isUsableWhen` can now make the context influence a change in the selected values (clearing it when not usable). The idea was that I'd watch for changes to these values, then update the consumer values, but it seems that it doesn't work unless I make the `useEffect` dependency the whole state object. Happy to just move this into attributes, but it does mean that I'd need another `reduce` to convert the `state.filters` object to the correct shape when passing back into `onValuesChange` as opposed to now just taking and passing it in as is.
- `(consumer) values` exists as the consumer may want to manipulate the values from an external event.

Reply:
Hm, probably not worth looking into shuffling this now then, your test coverage should help optimise it later if it becomes an issue.

Reply:
Yeah definitely :) And the only types I'm changing around for it is internal. If anyone digs for them and uses them, then they'll need to face the consequences of the "breaking" change.